kernel/src/engine/arrow_utils.rs

```diff
@@ -817,28 +863,42 @@ mod tests {
         buffer::{OffsetBuffer, ScalarBuffer},
     };
 
-    use crate::schema::{ArrayType, DataType, MapType, StructField, StructType};
+    use crate::schema::{
+        ArrayType, ColumnMetadataKey, DataType, MapType, MetadataValue, StructField, StructType,
+    };
+    use crate::table_features::ColumnMappingMode;
 
     use super::*;
 
     fn nested_parquet_schema() -> ArrowSchemaRef {
         Arc::new(ArrowSchema::new(vec![
-            ArrowField::new("i", ArrowDataType::Int32, false),
+            ArrowField::new("i", ArrowDataType::Int32, false).with_metadata(arrow_fid(1)),
             ArrowField::new(
                 "nested",
                 ArrowDataType::Struct(
                     vec![
-                        ArrowField::new("int32", ArrowDataType::Int32, false),
-                        ArrowField::new("string", ArrowDataType::Utf8, false),
+                        ArrowField::new("int32", ArrowDataType::Int32, false)
+                            .with_metadata(arrow_fid(4)),
+                        ArrowField::new("string", ArrowDataType::Utf8, false)
+                            .with_metadata(arrow_fid(5)),
                     ]
                     .into(),
                 ),
                 false,
-            ),
-            ArrowField::new("j", ArrowDataType::Int32, false),
+            )
+            .with_metadata(arrow_fid(3)),
+            ArrowField::new("j", ArrowDataType::Int32, false).with_metadata(arrow_fid(2)),
         ]))
     }
 
+    fn column_mapping_cases() -> [ColumnMappingMode; 2] {
+        [
+            ColumnMappingMode::Id,
+            ColumnMappingMode::Name,
+            // NOTE: Name and None are currently handled the same
```

Comment:
should we anyway test `None` in case they aren't in the future? love this way of reusing all the tests for the different styles btw, thanks